### ISSUE TITLE
validateOn custom events.

### ISSUE THREAD

**Issue by g5becks** [fabian-hiller/modular-forms#149]
When using web component libraries like ionic, it is common for the elements to fire non standard events. For example, ion-input https://ionicframework.com/docs/api/input#events

As far as I can see, the only way to work around this is to pass a validate function to the createFrom function, this is not ideal though. I'm not sure if it's in the plans to support ValidationMode typed in a way that allows events outside of , 'touched' | 'input' | 'change' | 'blur' | 'submit' - but I don't think that one can assume that these standard events are the only ones inputs will fire when considering all the different web component libraries.

**Comment by fabian-hiller** [comment]
Can you give me some more context? I'm not familiar with Ionic. What framework are you using? What exactly are the problems with using Modular Forms? What should Modular Forms do differently or better?

**Comment by g5becks** [comment]
The issue is not with modular-forms. I am not using ionic with a framework, just using @ionic/core with solid-js.

I have found that a few different web component libraries fire different events on user input that are not standard. The link I posted above shows the issue, ionic fires events like ionBlur, ionChange, ionInput , etc.  This makes modularforms almost impossible to use in an idiomatic way with these libraries. 

As I said though, this is more of a issue with those libraries as I see it than with modular-forms. It is something to consider though, as changing the type of Validation mode to

```type ValidationMode = 'touched' | 'input' | 'change' | 'blur' | 'submit' | string```

would easily solve this. But it is completely up to you and not an issue with the library if you choose not to.

Gonna close this as it is not really an issue with the library though.

Thanks.
